packages/ics721/src/execute.rs

```diff
@@ -258,8 +259,24 @@ where
         let instantiate = if CLASS_ID_TO_NFT_CONTRACT.has(deps.storage, class.id.clone()) {
             vec![]
         } else {
+            let class_id = ClassId::new(class.id.clone());
+            let salt = class_id.as_bytes();
+            // Get the canonical address of the contract creator
+            let canonical_creator = deps.api.addr_canonicalize(env.contract.address.as_str())?;
+            // get the checksum of the contract we're going to instantiate
+            let CodeInfoResponse { checksum, .. } = deps
+                .querier
+                .query_wasm_code_info(CW721_CODE_ID.load(deps.storage)?)?;
+            let canonical_cw721_addr = instantiate2_address(&checksum, &canonical_creator, salt)
+                .map_err(|_| StdError::generic_err("Could not calculate addr"))?;
+            let cw721_addr = deps.api.addr_humanize(&canonical_cw721_addr)?;
+
+            // Save classId <-> contract mappings.
+            CLASS_ID_TO_NFT_CONTRACT.save(deps.storage, class_id.clone(), &cw721_addr)?;
+            NFT_CONTRACT_TO_CLASS_ID.save(deps.storage, cw721_addr.clone(), &class_id)?;
+
             let message = SubMsg::<T>::reply_on_success(
-                WasmMsg::Instantiate {
+                WasmMsg::Instantiate2 {
```

Comment:
This is the main message that we are trying to get back in this PR right?

Reply:
Yes, ser